odoo/models.py

```diff
@@ -1992,10 +2002,14 @@ def _read_group_groupby(self, groupby_spec: str, query: Query) -> SQL:
                     "(date_trunc('week', %s::timestamp - INTERVAL %s) + INTERVAL %s)",
                     sql_expr, interval, interval,
                 )
+            elif spec := READ_GROUP_NUMBER_GRANULARITY.get(granularity):
+                sql_expr = SQL("date_part(%s, %s)", spec, sql_expr)
```

Comment:
Just wondering, shouldn't we treat `week_number` like `week` above, so that week numbers change on the right day?